clients/privacy-center/components/modals/consent-request-modal/ConsentRequestForm.tsx

```diff
@@ -42,16 +43,18 @@ const useConsentRequestForm = ({
 }) => {
   const identityInputs =
     config.consent?.button.identity_inputs ?? defaultIdentityInput;
+  const fidesUserDeviceId = useFidesUserDeviceIdCookie();
   const toast = useToast();
   const formik = useFormik({
     initialValues: {
-      email: "",
-      phone: "",
+      email: undefined,
+      phone: undefined,
```

Comment:
This was done to avoid the fields being sent in the eventual payload (as they were before when the defaults were `""`). React doesn't like it, because it throws a warning when the inputs are used now:

```
xt-dev.js:20 Warning: A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://reactjs.org/link/controlled-components
```

Reply:
yeah, this is an annoying thing about forms and react. we work around it by transforming `undefined` to `str`s before marshalling it into a form, and then un-marshalling it before submission. 

something like these two:

https://github.com/ethyca/fides/blob/1550545d8732e88b343f2a5b10958fdf2ac6e638/clients/admin-ui/src/features/common/form/inputs.tsx#L50-L56

https://github.com/ethyca/fides/blob/1550545d8732e88b343f2a5b10958fdf2ac6e638/clients/admin-ui/src/features/privacy-notices/form.ts#L22-L43

I can take a look later if you'd like, to see what makes sense here, just let me know!

Reply:
should be fixed by https://github.com/ethyca/fides/pull/3158/commits/8134418705dcaec8b2a3b3a35b2377dbdfe15f0c

I checked and the payload doesn't send the email/phone fields unless something is filled in 👍